src/tests/ftest/io/nvme_pool_capacity.py

```diff
@@ -0,0 +1,268 @@
+        """
+        Method Descripton:
+            This method is called with different test_cases.
+            Args:
+               num_pool (int): Total pools for running a test.
+            Returns:
+               None
+        """
+        num_jobs = self.params.get("no_parallel_job", '/run/ior/*')
+        # Create a pool
+        pool = {}
+
+        # Iterate through IOR different ior test sequence
+        for oclass, api, test, flags in product(self.ior_daos_oclass,
+                                                self.ior_apis,
+                                                self.ior_test_sequence,
+                                                self.ior_flags):
+            # Create the IOR threads
+            threads = []
+            for val in range(0, num_pool):
+                pool[val] = TestPool(self.context,
+                                     dmg_command=self.get_dmg_command())
+                pool[val].get_params(self)
+                # Split total SCM and NVME size for creating multiple pools.
+                pool[val].scm_size.value = int(float(test[0])) / num_pool
+                pool[val].nvme_size.value = int(float(test[1])) / num_pool
+                pool[val].create()
+                display_string = "pool{} space at the Beginning".format(val)
+                self.pool = pool[val]
+                self.pool.display_pool_daos_space(display_string)
+
+                for thrd in range(0, num_jobs):
+                    # Based on pools/jobs, split block size
+                    # if thrd == 0:
+                    #    tmp = int(float(test[3]) / num_pool)
+                    #    test[3] = str(tmp)
+                    # Add a thread for these IOR arguments
+                    threads.append(threading.Thread(target=self.ior_thread,
+                                                    kwargs={"pool": pool[val],
+                                                            "oclass": oclass,
+                                                            "api": api,
+                                                            "test": test,
+                                                            "flags": flags,
+                                                            "results":
+                                                            self.out_queue}))
+            # Launch the IOR threads
+            for thrd in threads:
+                self.log.info("Thread : %s", thrd)
+                thrd.start()
+                time.sleep(5)
+            # Wait to finish the threads
+            for thrd in threads:
+                thrd.join()
+
+            # Verify the queue and make sure no FAIL for any IOR run
+            # Test should fail with ENOSPC.
+            while not self.out_queue.empty():
+                if (self.out_queue.get() == "FAIL" and test[4] == "PASS") \
+                     or (self.out_queue.get() != "FAIL" and test[4] == "FAIL"):
+                    self.fail("FAIL")
```

Comment:
Is this "FAIL" a message? If so, it should be more descriptive like "Test failed with ENOSPC".

Reply:
No. This fail message is not related to ENOSPC... It is checking a condition where the queue has FAIL and the YAML file suggests a different output... It has nothing to do with ENOSPC,.